dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WebSocketPrivateStreamTestSuite.scala

```diff
@@ -0,0 +1,280 @@
+      val wsac = mkWsConnection(mkKeyPair("Test"), method)
+
+      wsac.getAllBalances should have size 1
+      wsac.getAllOrders should have size 0
+    }
+
+    "when user place and cancel limit order" in {
+      val acc = mkAccountWithBalance(150.usd -> usd)
+      val wsc = mkWsConnection(acc, method)
+
+      val bo1 = mkOrder(acc, wavesUsdPair, BUY, 100.waves, 100)
+      val bo2 = mkOrder(acc, wavesUsdPair, BUY, 10.waves, 100, version = 3, feeAsset = usd, matcherFee = 30)
+
+      placeAndWaitInStream(wsc, bo1)
+      placeAndWaitInStream(wsc, bo2)
+
+      eventually {
+        wsc.getAllBalances.size should be >= 3
+      }
+      wsc.getAllBalances should contain(usd   -> WsBalances(tradable = 150.0, reserved = 0.0))
+      wsc.getAllBalances should contain(Waves -> WsBalances(tradable = 0.0, reserved = 0.0))
+      wsc.getAllBalances should contain(usd   -> WsBalances(tradable = 39.7, reserved = 110.3))
+
+      wsc.getAllOrders.distinct should matchTo(
+        Seq(
+          WsOrder.fromDomain(LimitOrder(bo1), OrderStatus.Accepted),
+          WsOrder.fromDomain(LimitOrder(bo2), OrderStatus.Accepted)
+        )
+      )
+
+      wsc.clearMessagesBuffer()
+
+      cancelAndAwait(acc, bo1)
+
+      eventually {
+        wsc.getAllBalances.size should be >= 1
+      }
+      wsc.getAllBalances should contain(usd -> WsBalances(tradable = 139.7, reserved = 10.3))
+      wsc.getAllOrders should contain(WsOrder(bo1.id(), status = OrderStatus.Cancelled.name.some))
+
+      cancelAndAwait(acc, bo2)
+
+      eventually {
+        wsc.getAllOrders should contain(WsOrder(bo2.id(), status = OrderStatus.Cancelled.name.some))
+        wsc.getAllBalances should contain(usd -> WsBalances(tradable = 150.0, reserved = 0.0))
+      }
+    }
+
+    "when user place and fill market order" in {
+      val acc = mkAccountWithBalance(51.003.waves -> Waves)
+      val wsc = mkWsConnection(acc, method)
+      val smo = mkOrder(acc, wavesUsdPair, SELL, 50.waves, 100)
+
+      placeAndAwaitAtDex(mkOrder(alice, wavesUsdPair, BUY, 15.waves, 120))
+      placeAndAwaitAtDex(mkOrder(alice, wavesUsdPair, BUY, 25.waves, 110))
+      placeAndAwaitAtDex(mkOrder(alice, wavesUsdPair, BUY, 40.waves, 100))
+
+      dex1.api.placeMarket(smo)
+      waitForOrderAtNode(smo)
+
+      eventually {
+        wsc.getAllBalances should contain(Waves -> WsBalances(tradable = 51.003, reserved = 0.0))
+        wsc.getAllBalances should contain(Waves -> WsBalances(tradable = 1.0, reserved = 0.0))
+        wsc.getAllBalances should contain(usd   -> WsBalances(tradable = 55.5, reserved = 0.0))
+      }
+
+      wsc.getAllOrders.distinct should matchTo(
+        Seq(
+          WsOrder
+            .fromDomain(MarketOrder(smo, 0.waves), status = OrderStatus.Filled(0.waves, 0.waves))
+            .copy(
+              filledAmount = 50.0.some,
+              filledFee = 0.003.some,
+              avgWeighedPrice = 1.11.some // (15*120 + 25*110 + 10*100)/50
+            )
```

Comment:
`WsOrder.orderDiff`?

Reply:
no, orderDiff result is (None, None, None...) but there I need order + filled part

Reply:
I find a way to remove orderDif